managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1896,7 +1896,8 @@ void internalMarkDelete(final MarkDeleteEntry mdEntry) {
                 log.info("Skipping updating mark delete position to {}. The persisted mark delete position {} "
                         + "is later.", mdEntry.newPosition, persistentMarkDeletePosition);
             }
-            mdEntry.triggerComplete();
+            // run with executor to prevent deadlock
+            ledger.getExecutor().executeOrdered(ledger.getName(), safeRun(() -> mdEntry.triggerComplete()));
```

Comment:
I think that this behaviour is consistent with what happens a few lines below in the operationComplete() callback passed to `persistPositionToLedger`:` mdEntry.triggerComplete();` is not executed on this thread immediately but later

Reply:
Yes, it happened in a different thread that will not hold the `pendingMarkDeleteOps` lock, so will not encounter the deadlock.